.scripts/generate-sdks.ts

```diff
@@ -140,35 +159,102 @@ export async function findMissingSdks(azureRestApiSpecsRepository: string): Prom
-    const trimmedBuffer = readmeMdBuffer.slice(beginningIndex);
+async function getYamlSection(buffer: Buffer, sectionBeginning: string, sectionEnd: string): Promise<Buffer> {
+    const beginningIndex = buffer.indexOf(sectionBeginning);
+    const trimmedBuffer = buffer.slice(beginningIndex + (sectionBeginning.length));
 
-    const endIndex = trimmedBuffer.indexOf(sectionEnd);
+    const endIndex = trimmedBuffer.indexOf(sectionEnd, 3);
     const sectionBuffer = trimmedBuffer.slice(0, endIndex);
 
+    return sectionBuffer;
+}
+
+async function doesReadmeMdFileSpecifiesTypescriptSdk(readmeMdPath: string): Promise<boolean> {
+    const readmeMdBuffer = await fs.readFile(readmeMdPath);
+    const sectionBuffer = await getYamlSection(readmeMdBuffer, "``` yaml $(swagger-to-sdk)", "```");
+
     if (sectionBuffer.includes("azure-sdk-for-js")) {
         return true;
     }
 
     return false;
 }
 
-export async function copyExistingNodeJsReadme(sdkPath: string): Promise<void> {
+export async function copyExistingNodeJsReadme(sdkPath: string): Promise<string> {
     const nodeJsReadmePath = path.resolve(sdkPath, "readme.nodejs.md");
     const typescriptReadmePath = path.resolve(sdkPath, "readme.typescript.md");
 
     if (args.verbose) {
-        console.log(`Copying ${nodeJsReadmePath} to ${typescriptReadmePath}`)
+        _logger.log(`Copying ${nodeJsReadmePath} to ${typescriptReadmePath}`)
     }
 
     if (await exists(typescriptReadmePath)) {
         throw new Error(`${typescriptReadmePath} file already exists`)
     }
 
     await fs.copyFile(nodeJsReadmePath, typescriptReadmePath);
+    return typescriptReadmePath;
+}
+
+async function updatePackageName(settings: readmeSettings): Promise<readmeSettings> {
+    const packageName = settings.nodejs["package-name"]
+    if (packageName.startsWith("arm")) {
+        return settings;
+    }
+
+    if (!packageName.startsWith("azure-")) {
+        throw new Error("Unrecognized package name. Please update manually.");
+    }
+
+    settings.nodejs["package-name"] = packageName.replace("azure-", "");
+    return settings;
 }
+
+async function updateMetadataFields(settings: readmeSettings): Promise<readmeSettings> {
+    settings.nodejs["generate-metadata"] = true;
+    delete settings.nodejs["generate-license-txt"]
+    delete settings.nodejs["generate-package-json"]
+    delete settings.nodejs["generate-readme-md"];
+
+    return settings;
+}
+
+async function updateOutputFolder(settings: readmeSettings): Promise<readmeSettings> {
+    settings.nodejs["output-folder"] = `$(typescript-sdks-folder)/packages/${settings.nodejs["package-name"]}`;
+    return settings;
+}
+
+async function updateYamlSection(sectionText: string): Promise<string> {
+    const section = yaml.safeLoad(sectionText);
+    let updatedSection = await updatePackageName(section);
+    updatedSection = await updateMetadataFields(updatedSection);
+    updatedSection = await updateOutputFolder(updatedSection);
```

Comment:
There's no need to keep assigning to updatedSection, right? Since these functions just return the same object that they're given, then these are essentially assigning updatedSection it itself.